### PR TITLE
Add components to work with GCP

### PR DESCRIPTION
**Pull request by jlewi**
* Extend the Pachyderm Kubeflow package so that we can deploy it backed by GCS

* Related to #611

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/645)

### CODE REVIEW

File: kubeflow/pachyderm/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+```shell
+ks pkg install kubeflow/pachyderm@${VERSION}
+ks generate pachyderm pachyderm
+ks apply ${ENV} -c pachyderm
+```
+
+Once you've deployed pachyderm, follow pachyderm's 
+[user_guide](http://pachyderm.readthedocs.io/en/latest/getting_started/getting_started.html).
+You will need to install Pachyderm's CLI (pachctyl) and run `pachctl -n ${NAMESPACE} port-forward &`
```
Comment (dwhitena) [inline kubeflow/pachyderm/README.md]: Spelling. *pachctl

File: kubeflow/pachyderm/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+in order to interact with Pachyderm.
+
+**Warning** By default this local model of Pachyderm which uses ephmeral volumes
+to store data. It is not suitable for production use as data will be lost if the pods get  restarted.
```
Comment (dwhitena) [inline kubeflow/pachyderm/README.md]: Is this true for the GCP deploy as well?

File: kubeflow/pachyderm/all.libsonnet [hunk 3]
```diff
@@ -211,12 +347,15 @@
       },
     },
 
+    // TODO(jlewi): The local version(minikube) of Pachyderm runs etcd as a deployment.
+    // Why not always use a statefulset? Is because we need a storage class which may not be defined?
```
Comment (dwhitena) [inline kubeflow/pachyderm/all.libsonnet]: See comments from JD in the corresponding issue.

File: kubeflow/pachyderm/README.md (hunk 1 above)
Comment (jlewi) [reply kubeflow/pachyderm/README.md]: Done

File: kubeflow/pachyderm/README.md (hunk 2 above)
Comment (jlewi) [reply kubeflow/pachyderm/README.md]: No updated the doc to make this cleaner.

**Comment by jlewi** [comment]
Thanks @dwhitena PTAL

**Comment by dwhitena** [comment]
@jlewi LGTM. I might make a few formatting changes, but I can do that after getting this merged in. Info-wise, all looks good.

**Pull request merged**